graphql_compiler/query_pagination/__init__.py

```diff
@@ -4,14 +4,55 @@
 from graphql.language.printer import print_ast
 
 from ..ast_manipulation import safe_parse_graphql
-from ..cost_estimation.cardinality_estimator import estimate_number_of_pages
+from ..cost_estimation.analysis import QueryPlanningAnalysis
 from ..global_utils import ASTWithParameters, QueryStringWithParameters
 from ..schema.schema_info import QueryPlanningSchemaInfo
 from .pagination_planning import PaginationAdvisory
 from .query_splitter import split_into_page_query_and_remainder_query
 from .typedefs import PageAndRemainder
 
 
+def _estimate_number_of_pages(query: ASTWithParameters, result_size: float, page_size: int) -> int:
+    """Estimate how many pages of results will be generated for a given query.
+
+    Using the cardinality estimator, we generate an estimate for the query result cardinality i.e.
+    the number of result rows, then divide (rounding up) by the page_size to get the approximate
+    number of pages that the query will produce.
+    For example, if a query were estimated to return 12000 result rows, and the desired page size is
+    5000, then the query can be divided into ceil(12000/5000)=3 pages, each with a result size below
+    (or equal to) 5000 results.
+
+    Args:
+        query: ASTWithParameters
+        result_size: The estimated result size of a query
+        page_size: The desired page size of a query
+
+    Returns:
+        int, estimated number of pages if the query were executed.
+
+    Raises:
+        ValueError if page_size is below 1.
+    """
```

Comment:
Please rewrite docstring to account for functionality change and remove unused `query` parameter

Reply:
Will do. But `query` is used in error message